config/v1/types_infrastructure.go

```diff
@@ -507,6 +507,20 @@ type AWSPlatformStatus struct {
 	// +listType=atomic
 	// +optional
 	ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
+
+	// cloudLoadBalancerConfig is a union that contains the IP addresses of API,
+	// API-Int and Ingress Load Balancers created on the cloud platform. These
+	// values would not be populated on on-prem platforms. These Load Balancer
+	// IPs are used to configure the in-cluster DNS instances for API, API-Int
+	// and Ingress services. `dnsType` is expected to be set to `ClusterHosted`
+	// when these Load Balancer IP addresses are populated and used.
```

Comment:
This is good context but perhaps a little heavy on the dev side, perhaps we can reword a little to be more user oriented?

Perhaps we should fix the GCP version as well? 🤔 

Rather than API-Int, internal API load balancer. We can be more explicit when referring to the load balancers rather than inferring that the reader knows the internals of an openshift architecture.

I don't know that we need to state that this is a union, do users care?

When you say dnsType is expected to be set, is it not required to be set?

What about?
```suggestion
	// cloudLoadBalancerConfig holds configuration related to DNS and cloud
	// load balancers. It allows configuration of in-cluster DNS as an alternative
	// to the platform default DNS implementation.
	// When using the ClusterHosted DNS type, Load Balancer IP addresses
	// must be provided for the API and internal API load balancers as well as the
	// ingress load balancer.
```

Why did we settle on load balancer config when this is DNS? 🤔